src/views/Log.vue

```diff
@@ -503,6 +497,52 @@ export default {
         /* isGlobalCallback */ false
       )
     },
+    /**
+     * Return the default log file from the given log filenames, if there is a
+     * matching filename. Relies on the filenames having been sorted in descending
+     * order.
+     *
+     * @param {string[]} logFiles - list of available log filenames
+     * @returns {?string}
+     */
+    async getDefaultFile (logFiles) {
+      if (this.jobLog === 1) {
+        // we are looking at a job log => pick the appropriate log file based on its state
+        // get the job from the store
+        let result
+        try {
+          // get the list of available log files
+          result = await this.$workflowService.apolloClient.query({
```

Comment:
We've got an interface for performing queries like this. It's got a silly name and isn't documented or explained (sorry). The workflow service code is in need of refactor but rather low priority.

```suggestion
          result = await this.$workflowService.query2({
```

Reply:
fixed in ad47dc208af76bb2da0142a835d6dac5de5645ef